x-pack/plugins/security_solution/public/management/components/endpoint_response_actions_list/components/action_log_expanded_tray.tsx

```diff
@@ -0,0 +1,221 @@
+  &.euiDescriptionList {
+    > .euiDescriptionList__title {
+      color: ${(props) => props.theme.eui.euiColorDarkShade};
+      font-size: ${(props) => props.theme.eui.euiFontSizeXS};
+    }
+
+    > .euiDescriptionList__title,
+    > .euiDescriptionList__description {
+      font-weight: ${(props) => props.theme.eui.euiFontWeightRegular};
+      margin-top: ${(props) => props.theme.eui.euiSizeS};
+    }
+  }
+`;
+const topSpacingCss = css`
+  ${(props) => `${props.theme.eui.euiSize} 0`}
+`;
+const dashedBorderCss = css`
+  ${(props) => `1px dashed ${props.theme.eui.euiColorDisabled}`};
+`;
+const StyledDescriptionListOutput = euiStyled(EuiDescriptionList).attrs({ compressed: true })`
+  ${customDescriptionListCss}
+  dd {
+    margin: ${topSpacingCss};
+    padding: ${topSpacingCss};
+    border-top: ${dashedBorderCss};
+    border-bottom: ${dashedBorderCss};
+  }
+`;
+
+const StyledDescriptionList = euiStyled(EuiDescriptionList).attrs({
+  compressed: true,
+  type: 'column',
+})`
+  ${customDescriptionListCss}
+`;
+
+const StyledEuiCodeBlock = euiStyled(EuiCodeBlock).attrs({
+  transparentBackground: true,
+  paddingSize: 'none',
+})`
+  code {
+    color: ${(props) => props.theme.eui.euiColorDarkShade} !important;
+  }
+`;
+
+export const ActionsLogExpandedTray = memo<{
+  action: MaybeImmutable<ActionDetails>;
+}>(({ action }) => {
+  const getTestId = useTestIdGenerator('response-actions-list');
+  const { canWriteFileOperations, canWriteExecuteOperations } =
+    useUserPrivileges().endpointPrivileges;
+
+  const {
+    startedAt,
+    completedAt,
+    isCompleted,
+    wasSuccessful,
+    isExpired,
+    command: _command,
+    comment,
+    parameters,
+  } = action;
+
+  const parametersList = parameters
+    ? Object.entries(parameters).map(([key, value]) => {
+        return `${key}:${value}`;
+      })
+    : undefined;
+
+  const command = getUiCommand(_command);
+  const isGetFileCommand = command === 'get-file';
+  const isExecuteCommand = command === 'execute';
+  const dataList = [
+    {
+      title: OUTPUT_MESSAGES.expandSection.placedAt,
+      description: `${startedAt}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.startedAt,
+      description: `${startedAt}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.completedAt,
+      description: `${completedAt ?? emptyValue}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.input,
+      description: `${command}`,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.parameters,
+      description: parametersList ? parametersList : emptyValue,
+    },
+    {
+      title: OUTPUT_MESSAGES.expandSection.comment,
+      description: comment ? comment : emptyValue,
+    },
+  ].map(({ title, description }) => {
+    return {
+      title: <StyledEuiCodeBlock>{title}</StyledEuiCodeBlock>,
+      description: <StyledEuiCodeBlock>{description}</StyledEuiCodeBlock>,
+    };
+  });
+
+  const getOutputContent = () => {
```

Comment:
can you memoize it with `useCallback()`

Reply:
7a1f87b1bee40512a7d11b44ad16cd3d781748c6